src/fair/metrics.py

```diff
@@ -39,3 +39,156 @@ def leximin(X, agents, items):
     valuations.sort()
     valuations.reverse()
     return valuations
+
```

Comment:
I don't want this to majorly block you, but for new method, I'd really like to see

1. Type hints
2. Docstrings
3. Unit tests

In order to move this PR along, I'm ok with filing a separate issue for the unit tests. But do you mind please adding 1 and 2?